src/aioswitcher/device/tools.py

```diff
@@ -168,30 +171,40 @@ def convert_str_to_devicetype(device_type: str) -> DeviceType:
 
 def convert_token_to_packet(
     device_type: DeviceType, token: Union[str, None] = None
-) -> str:
-    """Return token packet from token if is valid, otherwise empty string."""
-    is_token_needed = bool(device_type.token_needed)
-    is_token_not_empty = token is not None and str(token) != ""
+) -> str | None:
+    """Convert a token to token packet.
+
+    Args:
+        device_type: the type of the device.
+        token: the token of the user sent by Email
+
+    Return:
+        Token packet if token is valid,
+        otherwise empty string or raise error.
+
+    """
+    token_key = b"jzNrAOjc%lpg3pVr5cF!5Le06ZgOdWuJ"
+    is_token_needed = device_type.token_needed
+    is_token_not_empty = bool(token)
     if is_token_needed:
-        if is_token_not_empty:
+        if is_token_not_empty and token is not None:
```

Comment:
The logic here seems strange, you call a function `convert_token_to_packet` which should return a token and the function itself check if the token is needed, it makes no sense, the call should check if the token is needed and decide if it should call this function, you are going to build a different packet/flow if token is needed so there is no use to check this inside here since you need to do this anyway on the caller side.

Reply:
It's not that stage if you can see where I am using it. [here](https://github.com/YogevBokobza/aioswitcher/blob/a193f88e52369401adce86ac98bc00736ad31bdc/src/aioswitcher/api/__init__.py#L113)
In this place, I want to set the token to be empty or token packet and "stop" the user there if there is an error

Reply:
It is not recommended to use a function that raise inside `__init__`, set the token there without any check, and when you create the packet raise if it is missing

Reply:
and as I wrote earlier this check is redundant since you do it anyhow when you use the token:
https://github.com/YogevBokobza/aioswitcher/blob/a193f88e52369401adce86ac98bc00736ad31bdc/src/aioswitcher/api/__init__.py#L192

Reply:
I see what you mean but the problem is that the convert function may raise an error when you are converting it if the token is not in the right format.. So I don't know at which time the conversion should happen.. 
Do you have an idea?

Reply:
> and as I wrote earlier this check is redundant since you do it anyhow when you use the token: https://github.com/YogevBokobza/aioswitcher/blob/a193f88e52369401adce86ac98bc00736ad31bdc/src/aioswitcher/api/__init__.py#L192

They are not that much the same.. in the place you just showed we need to validate that the token is needed and not empty so we can use the next packet for the API and the first is to get the token packet that can be a string or None or raise an error so we can use that in the next packet for the API

How would you approach this?

Reply:
Do you think that instead of [this](https://github.com/YogevBokobza/aioswitcher/blob/a193f88e52369401adce86ac98bc00736ad31bdc/src/aioswitcher/api/__init__.py#L113)
it will be better to not have `self._token` and have the token_packet as a normal variable that is defined in the operations API like `set_position` and use that instead of `is_token_valid`?

Reply:
> Do you think that instead of [this](https://github.com/YogevBokobza/aioswitcher/blob/a193f88e52369401adce86ac98bc00736ad31bdc/src/aioswitcher/api/__init__.py#L113) it will be better to not have `self._token` and have the token_packet as a normal variable that is defined in the operations API like `set_position` and use that instead of `is_token_valid`? (In this approach a conversation would have to happen every time..)

I think we should discuss this next PR when I will show you the usage of the convert token to token packet..

Reply:
Are you referring to this?
```
src/aioswitcher/device/tools.py:194: error: Argument 1 to "bytes" has incompatible type "str | None"; expected "str"  [arg-type]
```

Reply:
> I think we should discuss this next PR when I will show you the usage of the convert token to token packet..

So please fix it for now and when we see the usage in the next PR we can discuss it